### PR TITLE
Can add supplementary files to debian control

### PR DESCRIPTION
**Pull request by eskatos**
Regarding #109 here is a PR allowing to add arbitrary files to the debian control directory.
For example, using this, one is able to put a `changelog` file in there.

Please tell me if you're willing to merge this and if more changes are needed.

Cheers

### CODE REVIEW

File: src/test/groovy/com/netflix/gradle/plugins/deb/DebPluginTest.groovy [hunk 1]
```diff
@@ -844,4 +844,29 @@ class DebPluginTest extends ProjectSpec {
         appleFile.userName == 'me'
         appleFile.groupName == 'awesome'
     }
+
```
Comment (rspieldenner) [inline src/test/groovy/com/netflix/gradle/plugins/deb/DebPluginTest.groovy]: have a test for supplementaryControl null which throws an error

File: src/main/groovy/com/netflix/gradle/plugins/deb/DebCopyAction.groovy [hunk 2]
```diff
@@ -278,6 +278,11 @@ class DebCopyAction extends AbstractPackagingCopyAction {
                 }
         debianFiles.addAll(addlFiles)
 
+        (debTask.allSupplementaryControlFiles - null).each { supControl ->
```
Comment (rspieldenner) [inline src/main/groovy/com/netflix/gradle/plugins/deb/DebCopyAction.groovy]: Do you need the - null? Can we just make sure null don't get in here?

**Comment by eskatos** [comment]
Hey @rspieldenner, thanks for your feedback.

I just pushed some changes following your remarks.

I removed the `- null` in `DepCopyAction` by properly adding lists in `SystemPackagingTask`. Before f9102db the list always had a `null` element because groovy does `[] + null = [null]`.
I have the gut feeling that every other collections in `SystemPackagingTask` suffer the very same issue but it looks like it is not exercised.

As for the `supplementaryControl null` test, I'm not sure it is needed as every other collections taking either a String or a File in `SystemPackagingExtension`, including `supplementaryControlFiles`, behave the same way when given `null`: they throw a `GroovyRuntimeException` with the following reason: "Cannot resolve which method to invoke for [null] due to overlapping prototypes".

Here would be a test for this, not included in the commits, tell me if I should:

```groovy
    @FailsWith(value = GroovyRuntimeException)
    def "Throw error on null supplementary file added to debian control"() {
        given:
        project.apply plugin: 'deb'
        Deb debTask = project.task('buildDeb', type: Deb) {
            packageName = 'throw-error-on-null-supplementary-file-added-debian-control'
            supplementaryControl null
        }
        when:
        debTask.execute()
        then:
        "should fail"
    }
```

**Comment by eskatos** [comment]
Tell me if you need me to squash the commits in a single one.

**Comment by eskatos** [comment]
Any feedback on this PR?

**Comment by rspieldenner** [comment]
Looks good now.

**Comment by eskatos** [comment]
Cool, thanks!

**Comment by eskatos** [comment]
With this added, one can now easily tailor debian control directory content.
Looking forward next available release.

Cheers :beers:

**Pull request merged**